kernel/src/main/java/org/kframework/parser/json/JsonParser.java

```diff
@@ -349,16 +362,21 @@ private static K toK(JsonObject data) throws IOException {
 
             case KAPPLY:
                 int arity = data.getInt("arity");
-                K[] args  = toKs(arity, data.getJsonArray("args"));
+                K[] args  = toKs(arity, data.getJsonArray("args"), mod);
                 label     = data.getString("label");
-                klabel    = data.getBoolean("variable")
-                          ? KVariable(label)
-                          : KLabel(label);
+                if (data.getBoolean("variable"))
+                    klabel = KVariable(label);
+                else {
+                    if (mod.productionsForKLabelName().get(label).isDefined())
+                        klabel = mod.productionsForKLabelName().get(label).get().iterator().next().klabel().get();
```

Comment:
Should we enforce here that we get a unique label? What does it mean if there are multiple?

Reply:
Maybe we should build into `productionForKLabelName` that it only gets one label back in each case. And if there are places in the prelude where it's not true, then we special-case them. I think that `.` and `.K` and `.::K` may be examples where it's not true.